extensions/pkg/controller/extension/reconciler.go

```diff
@@ -168,15 +168,21 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 		return reconcile.Result{}, err
 	}
 
-	var (
-		result reconcile.Result
-		err    error
-	)
+	var result reconcile.Result
 
 	if ex.DeletionTimestamp != nil {
 		return r.delete(r.ctx, ex)
 	}
 
+	shoot, err := extensionscontroller.GetShoot(r.ctx, r.client, request.Namespace)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if extensionscontroller.IsShootFailed(shoot) {
+		return reconcile.Result{}, nil
+	}
```

Comment:
Shouldn't this be moved up to line 172?

Reply:
[same question as here](https://github.com/gardener/gardener/pull/2279#discussion_r419915471)